source/common/http/header_map_impl.cc

```diff
@@ -22,6 +22,7 @@ namespace {
 
 constexpr absl::string_view DelimiterForInlineHeaders{","};
 constexpr absl::string_view DelimiterForInlineCookies{"; "};
+const static int kMinHeadersForLazyMap = 3; // Arbitrary hard-coded value.
```

Comment:
I think it's better than arbitrary :) @adisuissa did ran microbenchmarks to at least sanity-check that choice.

Reply:
3 was chosen after doing some benchmarks (#12656).